nixos/modules/system/boot/tmp.nix

```diff
@@ -3,62 +3,60 @@
 with lib;
 
 let
-  cfg = config.boot;
+  cfg = config.boot.tmp;
 in
 {
-
-  ###### interface
+  imports = [
+    (mkRenamedOptionModule [ "boot" "cleanTmpDir" ] [ "boot" "tmp" "cleanOnBoot" ])
+    (mkRenamedOptionModule [ "boot" "tmpOnTmpfs" ] [ "boot" "tmp" "onTmpfs" ])
+    (mkRenamedOptionModule [ "boot" "tmpOnTmpfsSize" ] [ "boot" "tmp" "tmpfsSize" ])
+  ];
 
   options = {
-
-    boot.cleanTmpDir = mkOption {
+    boot.tmp.cleanOnBoot = mkOption {
       type = types.bool;
       default = false;
       description = lib.mdDoc ''
         Whether to delete all files in {file}`/tmp` during boot.
       '';
     };
 
-    boot.tmpOnTmpfs = mkOption {
+    boot.tmp.onTmpfs = mkOption {
       type = types.bool;
       default = false;
       description = lib.mdDoc ''
          Whether to mount a tmpfs on {file}`/tmp` during boot.
```

Comment:
Maybe we can use this opportunity to also add inform about side effects, such as large builds (like the kernel) failing if there isn't enough space?

Last time I tried, a 16GB tmpfs wasn't enough space to build a kernel using Nix.

Reply:
If you have a suggestion feel free to provide an improved text.